src/lib/config.ts

```diff
@@ -1296,9 +1296,10 @@ export async function save(storage: "local" | "sync" = get("storageloc")) {
     // storageobj.set({CONFIGNAME: USERCONFIG})
     const settingsobj = o({})
     settingsobj[CONFIGNAME] = USERCONFIG
-    return storage === "local"
+    storage === "local"
         ? browser.storage.local.set(settingsobj)
         : browser.storage.sync.set(settingsobj)
+    return new Promise(resolve => setTimeout(resolve, 0))
```

Comment:
Quoting @farnoy from Matrix:

> I believe you can "fix" config.set by adding await new Promise(resolve => setTimeout(resolve, 0)) after this line https://github.com/tridactyl/tridactyl/pull/2208/files#diff-3d46ae9b8655de6d492285665a944ff8R69
this should effectively wait until onChanged handlers get dispatched before returning control